src/uct/ib/ud/base/ud_iface.h

```diff
@@ -128,9 +128,10 @@ struct uct_ud_iface {
         ucs_queue_head_t       resend_skbs;
         unsigned               resend_skbs_quota;
         ucs_arbiter_t          pending_q;
-        int                    pending_q_len;
-        int                    in_pending;
         ucs_queue_head_t       async_comp_q;
+        /* true if the pending_q could have been dispatched from the
+         * async progress */
+        uint8_t                pending_async_ev;
```

Comment:
1. move this before "int16_t available" to compact the struct
2. comment: /* got async events but pending queue was not dispatched */
3. maybe rename to "async_before_pending"?